File: Reverse_Linked_List_Iterative.cpp
Iterative method
Time Complexity-O(n)
Space Complexity-O(1)
Did the code run on Leetcode? yes

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *p=head,*a=NULL,*b=NULL;
        while(p!=NULL)
        {
            a=p->next;
            p->next=b;
            b=p;
            p=a;
        }
        head=b;
        return b;
        
    }
};

File: reverse-linked-list-iterative.java
//Time complexity : O(n)
//Space complexity : O(1)
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        //curr pointer will point to head
        ListNode curr = head;
        while(curr != null) {
            //temp pointer will point to curr.next
            ListNode temp = curr.next;
            //curr.next point now will point to previous
            curr.next = prev;
            prev = curr;
            curr = temp;
        }
        return prev;
    }
}